src/pool/cli.c

```diff
@@ -892,9 +893,14 @@ pool_connect_cp(tse_task_t *task, void *data)
 		D_GOTO(out, rc);
 	}
 
+	rs = &pco->pco_rebuild_st;
 	rc = process_query_reply(tpriv->pool, map_buf, pco->pco_op.po_map_version,
-				 pco->pco_op.po_hint.sh_rank, &pco->pco_space, &pco->pco_rebuild_st,
-				 NULL /* tgts */, info, NULL, NULL, true);
+				 pco->pco_op.po_hint.sh_rank, &pco->pco_space, rs, NULL /* tgts */,
+				 info, NULL, NULL, true);
+
+	tpriv->pool->dp_max_supported_layout_ver = rs->rs_max_supported_layout_ver
+						       ? rs->rs_max_supported_layout_ver
+						       : DAOS_POOL_OBJ_VERSION_1;
```

Comment:
If there is a DAOS_PROP_PO_OBJ_VERSION and it's VERSION_2 or above, we should still use it right? So we can remove this hack after a couple of rounds of upgrades .

Reply:
If rs_max_supported_layout_ver is not 0, then it should be >= VERSION_2. we will use it directly. rs_max_supported_layout_ver is 0 means it is old server